stablehlo/reference/ProcessGrid.h

```diff
@@ -154,12 +172,38 @@ class ProcessGrid {
   /// StableHLO `outfeed` represented as a queue.
   std::queue<SmallVector<Tensor>> outfeed_;
 
+  /// Synchronization primitive used to manage concurrent access to `outfeed_`.
   std::mutex outfeedLock_;
 
   /// Synchronization primitive used to manage concurrent access to
   /// `channelLocks_`.
   std::mutex rendezvousLock_;
 
+  /// Synchronization primitive used to manage concurrent access to
+  /// `sendRecvChannels_`.
+  std::mutex sendRecvOmniLock_;
+
+  /// Internal storage used to implement `send` and `recv`.
+  /// `send` can write its data to the channel with ChannelId once the ops are
+  /// ready to communicate. `recv` receives data from the same channel once the
+  /// data is ready to read.
+  std::map<ChannelId, RendezvousResult> sendRecvChannels_;
+
+  /// Synchronization primitive used to manage concurrent access to
+  /// `sendRecvChannels_`.
+  std::map<ChannelId, std::mutex> sendRecvChannelLocks_;
```

Comment:
Should any of these be `ThreadSafeMap`?

Reply:
Rebased to HEAD and updated the commit with recent addition of `ThreadSafeMap`. PTAL.